### PR TITLE
Added method nearestPoints and refactored nearestPoint to use it

### PR DESCRIPTION
**Pull request by alexisbar**
Hello,
The nearestPoint method added in PR #18 wasn't appropriate for my use-case, as it returned only the first point of GEOSNearestPoints_r(), which itself is returning an array of points. 

I added a nearestPoints (with a S) method returning an array of Coordinate as specified in this GEOS spec http://geos.refractions.net/ro/doxygen_docs/html/classgeos_1_1operation_1_1distance_1_1DistanceOp.html#a6

To avoid breaking everything, I refactored the nearestPoint (without the S) which use my method. Don't know if it's the best to do, feel free to tell me.

### CODE REVIEW

**Comment by vfn** [comment]
@alexisbar That's great! Could you add some tests as well?

**Comment by vfn** [comment]
@alexisbar I noticed you added another commit to the PR, but I can only see spaces being added, but no tests at all https://github.com/GEOSwift/GEOSwift/pull/72/files/90e2609c047f127f47ce4407464067d972a636f0..be8c0371eec2b792bf850c5bed0e635d3af26c08

**Comment by alexisbar** [comment]
@vfn done !

File: GEOSwiftTests/GEOSTests.swift [hunk 1]
```diff
@@ -97,15 +97,15 @@ class GEOSwiftTests: XCTestCase {
                 result = true
             }
         }
-        XCTAssert(result, "WKT parse failed (expected to receive a MULTIPOINT)")
```
Comment (vfn) [inline GEOSwiftTests/GEOSTests.swift]: Did you mean to remove that?

File: GEOSwiftTests/GEOSTests.swift [hunk 2]
```diff
@@ -10,7 +10,7 @@ import XCTest
 import GEOSwift
 
 class GEOSwiftTests: XCTestCase {
-
```
Comment (vfn) [inline GEOSwiftTests/GEOSTests.swift]: Would you be able to discard these changes when committing?

File: GEOSwiftTests/GEOSTests.swift [hunk 3]
```diff
@@ -114,4 +114,22 @@ class GEOSwiftTests: XCTestCase {
...
     }
+    
+    func testNearestPoints() {
+        let point = Geometry.create("POINT(45 9)") as! Waypoint
+        let polygon = Geometry.create("POLYGON((35 10, 45 45, 15 40, 10 20, 35 10),(20 30, 35 35, 30 20, 20 30))") as! Polygon
+        
+        let arrNearestPoints = point.nearestPoints(polygon)
+        
+        XCTAssertNotNil(arrNearestPoints, "Failed to get nearestPoints array between the two geometries")
+
```
Comment (vfn) [inline GEOSwiftTests/GEOSTests.swift]: Assert also that the number of expected points is 2

File: GEOSwiftTests/GEOSTests.swift [hunk 4]
```diff
@@ -114,4 +114,22 @@ class GEOSwiftTests: XCTestCase {
...
+    
+    func testNearestPoints() {
+        let point = Geometry.create("POINT(45 9)") as! Waypoint
+        let polygon = Geometry.create("POLYGON((35 10, 45 45, 15 40, 10 20, 35 10),(20 30, 35 35, 30 20, 20 30))") as! Polygon
+        
+        let arrNearestPoints = point.nearestPoints(polygon)
+        
+        XCTAssertNotNil(arrNearestPoints, "Failed to get nearestPoints array between the two geometries")
+        
+        var result = false
```
Comment (vfn) [inline GEOSwiftTests/GEOSTests.swift]: Please create individual asserts for each pair.
```
XCTAssertEqual(arrNearestPoints[0].x, point.nearestPoint(polygon).x)
XCTAssertEqual(arrNearestPoints[0].y, point.nearestPoint(polygon).y)
```

**Comment by alexisbar** [comment]
Changes done, thanks for your feedback

**Comment by vfn** [comment]
@alexisbar thank you! I have merged your changes and I'll be pushing a new version of the pod. Cheers!

**Pull request merged**